### PR TITLE
Fix multiple subscribes

### LINKED ISSUE #13: Subscribe is broken for multiple topics
The documentation suggests it should be possible to subscribe to more than one topic with the same connection, which is sort of true, however all of the subscriptions now point to the last function meaning it's impossible to send different topics to different functions without writing custom dispatch code.

The following I would expect messages to test.one to print out "1: my message", but it actually prints "3: my message", as do messages to test.two and test.thr

```
(defn testsubs []
  (doto (mh/connect "tcp://127.0.0.1:1883" (mh/generate-id))
    (mh/subscribe ["test.one"] (fn [_ _ data] (println "1: " data)))
    (mh/subscribe ["test.two"] (fn [_ _ data] (println "2: " data)))
    (mh/subscribe ["test.thr"] (fn [_ _ data] (println "3: " data)))))
```

### PR DESCRIPTION
**Pull request by jakubholynet**
Fix #13 while also:

- refactor `connect` to take `uri` + map of options
- moving `.setCallback` to `connect`
- fix tests to use topic names that don't conflict with RabbitMQ

### CODE REVIEW

**Comment by jakubholynet** [comment]
@michaelklishin I would be glad to hear your feedback, especially on the refactoring of `connect`. It might be somewhat confusing that the connect's options map has also the key `:opts`. Perhaps it would be better to call it `:mqtt-opts` or something .... ?

**Comment by jakubholynet** [comment]
I see that while the tests run locally, they fail in Travis with

> actual: org.eclipse.paho.client.mqttv3.MqttException: Too many publishes in progress

Not sure what to do. Add some wait time or decrease number of attempts?

**Comment by michaelklishin** [comment]
@jakubholynet that's because Paho Java is IoT scale and it intentionally fails when there are too many messages in flight. Unless we can control the threshold we have to reduce the number of attempts, yes.

**Comment by jakubholynet** [comment]
@michaelklishin FYI I will be offline between January 1st and 16th.

**Comment by jakubholynet** [comment]
Regarding the failing test(s) - perhaps we could simply `setMaxInflight(something greater that the default 10)`?

**Comment by michaelklishin** [comment]
@jakubholynet agreed, that's a better approach. We should set it to something like 10K by default in Machine Head. 10 is laughable even for development environments.

**Comment by michaelklishin** [comment]
@jakubholynet enjoy your holiday break! Is there anything left to be done in this PR or the docs or should I try to give it one final look tomorrow (conveniently I have 10 or so hours to spend at an airport during a layover)?

File: test/clojurewerkz/machine_head/client_test.clj [hunk 1]
```diff
@@ -13,10 +13,13 @@
 
 (def ci? (System/getenv "CI"))
 
+(defn connect-high-traffic []
+  (mh/connect "tcp://127.0.0.1:1883" {:opts {:max-inflight 1000}}))
```
Comment (michaelklishin) [inline test/clojurewerkz/machine_head/client_test.clj]: Please rename this to just `connect` and use a value of 50K or so. There is no downside to it (at least for CI) that I can think of.

**Comment by jakubholynet** [comment]
> Is there anything left to be done in this PR or the docs

Not from my side. Unless you have some improvement suggestions regarding the names of the `:opts` argument or anything else, you can just merge it in.

**Comment by michaelklishin** [comment]
@jakubholynet 10-4.

**Comment by michaelklishin** [comment]
@jakubholynet thanks. I ended up renaming one option so far. I also don't see any PRs updating the docs (https://github.com/clojurewerkz/machine_head.docs), is this planned?

**Comment by jakubholynet** [comment]
Yes, I plan to go through the docs.

**Pull request merged**